### PR TITLE
Generate API docs in @daffodil packages using dgeni

### PR DESCRIPTION
**Pull request by lderrickable**
## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] The commit message follows our guidelines: https://github.com/graycoreio/daffodil/blob/master/CONTRIBUTING.md#commit
- [x] Tests for the changes have been added (for bug fixes / features)
- [x] Docs have been added / updated (for bug fixes / features)

## PR Type
What kind of change does this PR introduce?
```
[ ] Bugfix
[ ] Feature
[ ] Code style update (formatting, local variables)
[ ] Refactoring (no functional changes, no api changes)
[x] Build related changes
[ ] CI related changes
[x] Documentation content changes
[ ] Other... Please describe:
```

## What is the current behavior?
We don't have api document generation right now.

Fixes: N/A

## What is the new behavior?
Documentation can now be generated from all @daffodil packages using dgeni. 

## Does this PR introduce a breaking change?
```
[ ] Yes
[x] No
```

### CODE REVIEW

File: tools/dgeni/karma.conf.js [hunk 1]
```diff
@@ -0,0 +1,31 @@
+// Karma configuration file, see link for more information
+// https://karma-runner.github.io/1.0/config/configuration-file.html
+
+module.exports = function (config) {
```
Comment (damienwebdev) [inline tools/dgeni/karma.conf.js]: I don't think we run tests in the browser, so I don't think this file does anything?

File: tools/dgeni/src/test.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+// This file is required by karma.conf.js and loads recursively all the .spec and framework files
+
+import 'core-js/es7/reflect';
```
Comment (damienwebdev) [inline tools/dgeni/src/test.ts]: I don't think we need this file for the tests either.

File: tools/dgeni/karma.conf.js (hunk 1 above)
Comment (lderrickable) [reply tools/dgeni/karma.conf.js]: you're right. Removed

File: tools/dgeni/src/test.ts (hunk 2 above)
Comment (lderrickable) [reply tools/dgeni/src/test.ts]: Removed

**Pull request merged**